src/overlays/actors/ovl_En_Time_Tag/z_en_time_tag.c

```diff
@@ -42,40 +42,268 @@ const ActorInit En_Time_Tag_InitVars = {
+                }
+            }
+            break;
+
+        case 2:
+            this->actionFunc = func_80ACA5F8;
+            break;
+    }
+
+    if (this->actor.home.rot.x != 0) {
+        if (this->actor.cutscene == -1) {
+            this->actor.home.rot.x = 0;
+        } else if (ActorCutscene_GetCurrentIndex() == 0x7C) {
+            ActorCutscene_Stop(0x7C);
+            ActorCutscene_SetIntentToPlay(this->actor.cutscene);
+        } else if (ActorCutscene_GetCanPlayNext(this->actor.cutscene)) {
+            ActorCutscene_Start(this->actor.cutscene, &this->actor);
+            this->actor.home.rot.x = 0;
+        } else {
+            ActorCutscene_SetIntentToPlay(this->actor.cutscene);
+        }
+    }
+}
+
+void func_80ACA5F8(EnTimeTag* this, GlobalContext* globalCtx) {
+    if (Actor_ProcessTalkRequest(&this->actor, &globalCtx->state)) {
+        if (gSaveContext.save.playerForm == PLAYER_FORM_ZORA) {
+            if ((ENTIMETAG_GET_SWITCHFLAG(&this->actor)) == 1) {
+                Message_StartTextbox(globalCtx, 0x101C, &this->actor);
+            } else {
+                Message_StartTextbox(globalCtx, 0x122B, &this->actor);
+            }
+            this->actor.home.rot.x = 1;
+        } else {
+            Message_StartTextbox(globalCtx, 0x122A, &this->actor);
+            if (0) {}
+            ((EnElf*)GET_PLAYER(globalCtx)->tatlActor)->unk_264 |= 4;
+            Actor_ChangeFocus(&this->actor, globalCtx, GET_PLAYER(globalCtx)->tatlActor);
+            this->actor.home.rot.x = 0;
+        }
+        this->actionFunc = func_80ACA418;
+    } else if ((this->actor.xzDistToPlayer < 100.0f) && Player_IsFacingActor(&this->actor, 0x3000, globalCtx)) {
+        func_800B8614(&this->actor, globalCtx, 110.0f);
+    }
+}
+
+void func_80ACA714(EnTimeTag* this, GlobalContext* globalCtx) {
+}
+
+void func_80ACA724(EnTimeTag* this, GlobalContext* globalCtx) {
+    if (Message_GetState(&globalCtx->msgCtx) == 5) {
+        globalCtx->nextEntranceIndex = globalCtx->setupExitList[ENTIMETAG_GET_PATH(&this->actor)];
+        globalCtx->sceneLoadFlag = 0x14;
+        if (!(ENTIMETAG_GET_E000(&this->actor))) {
+            Actor_PlaySfxAtPos(&this->actor, NA_SE_OC_DOOR_OPEN);
+        }
+        this->actionFunc = func_80ACA714;
+    }
+}
+
+void func_80ACA7C4(EnTimeTag* this, GlobalContext* globalCtx) {
+    if (!(gSaveContext.save.weekEventReg[63] & 1) && !(gSaveContext.save.weekEventReg[63] & 2)) {
+        func_800B7298(globalCtx, &this->actor, 7);
+        Message_StartTextbox(globalCtx, ENTIMETAG_GET_1FE0(&this->actor) + 0x1883, NULL);
+        this->actionFunc = func_80ACA724;
+    }
+}
+
+void func_80ACA840(EnTimeTag* this, GlobalContext* globalCtx) {
+    s16 temp_ft4;
+    s16 temp_hi;
+
+    if ((globalCtx->sceneNum != SCENE_YADOYA) || (INV_CONTENT(ITEM_ROOM_KEY) != ITEM_ROOM_KEY)) {
+        temp_ft4 = gSaveContext.save.time * 0.00036621094f;
+        temp_hi = (s32)(gSaveContext.save.time * 0.021972656f) % 60;
```

Comment:
```suggestion
        temp_ft4 = gSaveContext.save.time * (24.0f / 0x10000); // TIME_TO_HOURS_F
        temp_hi = (s32)TIME_TO_MINUTES_F(gSaveContext.save.time) % 60;
```
These floats are time conversions. 
Hmm, this is the first time I've seen an actual `TIME_TO_HOURS_F` possible macro